src/types/json.h

```diff
@@ -179,6 +179,33 @@ struct JsonValue {
     return result_count;
   }
 
+  StatusOr<std::vector<std::optional<uint64_t>>> ArrInsert(std::string_view path, const int64_t &index,
+                                                           const std::vector<jsoncons::json> &insert_values) {
+    std::vector<std::optional<uint64_t>> result_count;
+
+    try {
+      jsoncons::jsonpath::json_replace(
+          value, path, [&insert_values, &result_count, index](const std::string & /*path*/, jsoncons::json &val) {
+            if (val.is_array()) {
+              auto len = static_cast<int64_t>(val.size());
+              if (index >= len /*index > 0*/ || len + index < 0 /*index < 0*/) {
```

Comment:
```suggestion
              if (index >= len /*index is out of upper bound*/ || len + index < 0 /*index < 0*/) {
```

Reply:
It is not typo, It is mean when index > 0.
I think I can change this line to ` if (index >= len /*when index > 0*/ || len + index < 0 /*when index < 0*/) {`

Reply:
`/*when index > 0*/` it should be >= length instead > 0? since index > 0 is good if it doesn't exceed the upper bound of the array.

Reply:
I means that when index > 0, we should check if index >= len. And when index < 0, we should check if index + len < 0.
It seems that this comment could be easily misunderstood. Should I delete it?

Reply:
oh, I did misunderstand your point. 

> It seems that this comment could be easily misunderstood. Should I delete it?

Sounds good, or add a longer comment line outside to make it clearer.

Reply:
Ok, I have added a longer comment above.